### PR TITLE
Updated iOS App Transport Security (ATS) general checks, domain exceptions and reporting

### PR DESCRIPTION
**Pull request by rajuraju14**
### Updated iOS ATS Checks and Reporting

```
Updated/added code to reflect Apple's iOS general ATS checks and domain exceptions as documented on their developer website. The following new properties/keys have been added:

- NSAllowsLocalNetworking
- NSExceptionAllowsInsecureHTTPLoads
- NSExceptionMinimumTLSVersion
- NSExceptionRequiresForwardSecrecy
- NSRequiresCertificateTransparency

Reporting templates have been updated as well.

```

### Checklist for PR

- [ ] Run MobSF unit tests and lint `tox -e lint,test`.
- [ ] Tested Working on Linux, Mac, Windows, and Docker
- [ ] Make sure build is passing on your PR. [Build Status](https://travis-ci.com/MobSF/Mobile-Security-Framework-MobSF/pull_requests)

### CODE REVIEW

File: templates/pdf/ios_report.html [hunk 1]
```diff
@@ -217,16 +217,40 @@ <h2><i class="fas fa-lock"></i> APP TRANSPORT SECURITY (ATS)</h2>
...
-                        <td>
-                         {{findings.issue}}
-                        <td>
-                          <span class="danger"> {{findings.status}}</span>
-                        </td>
-                        <td>
-                         {{findings.description}}
-                        </td>
-                      </tr>
+                        {% if "insecure" in findings.status %}
```
Comment (ajinabraham) [inline templates/pdf/ios_report.html]: `{% if  findings.status =="insecure"  %}` is a non ambiguous check.

Comment (ajinabraham) [reply templates/pdf/ios_report.html]: There is unnecessary repeating of description and issue.

I would replace 
```  
<td>
     <span class="danger"> {{findings.status}}</span>
</td>
```
with

```
<td>
{% if findings.status =="insecure" %}
<span class="danger"> {{findings.status}}</span>
{% elif findings.status =="secure" %}
<span class="success"> {{findings.status}}</span>
{% elif findings.status =="warning" %}
<span class="warning"> {{findings.status}}</span>
{% elif findings.status =="info" %}
<span class="info"> {{findings.status}}</span>
{% endif %}
</td>
```
This repeats less things and is a balanced approach. You can also do the if condition inside span class. But this is more readable.

File: templates/static_analysis/ios_binary_analysis.html [hunk 2]
```diff
@@ -506,16 +506,40 @@ <h5 class="card-title"></h5>
...
-                        <td>
-                         {{findings.issue}}
-                        <td>
-                          <span class="badge bg-danger"> {{findings.status}}</span>
-                        </td>
-                        <td>
-                         {{findings.description}}
-                        </td>
-                      </tr>
+                        {% if "insecure" in findings.status %}
```
Comment (ajinabraham) [inline templates/static_analysis/ios_binary_analysis.html]: See my comments in `ios_report.html`

File: templates/static_analysis/ios_source_analysis.html [hunk 3]
```diff
@@ -518,16 +518,40 @@ <h5 class="card-title"></h5>
...
-                        <td>
-                         {{findings.issue}}
-                        <td>
-                          <span class="badge bg-danger"> {{findings.status}}</span>
-                        </td>
-                        <td>
-                         {{findings.description}}
-                        </td>
-                      </tr>
+                        {% if "insecure" in findings.status %}
```
Comment (ajinabraham) [inline templates/static_analysis/ios_source_analysis.html]: See my comments in `ios_report.html`

File: StaticAnalyzer/views/ios/app_transport_security.py [hunk 4]
```diff
@@ -32,46 +32,131 @@ def check_transport_security(p_list):
...
+                            'or to be able to loosen the '
+                            'server trust evaluation '
+                            'requirements for HTTPS '
+                            'connections to the domain.'.format(domain)
+                        ),
+                    }
+                    ats.append(findings)
+
+                includes_subdomains = config.get('NSIncludesSubdomains', False)
+                if includes_subdomains is True:
```
[the remark this answers is not in the document]
Comment (rajuraju14) [reply StaticAnalyzer/views/ios/app_transport_security.py]: This has been resolved in the latest commit.

File: StaticAnalyzer/views/ios/app_transport_security.py [hunk 5]
```diff
@@ -32,46 +32,131 @@ def check_transport_security(p_list):
...
+                        'issue': ('NSIncludesSubdomains set to TRUE'
+                                  ' for {}'.format(domain)),
+                        'status': 'insecure',
+                        'description': (
+                            'NSIncludesSubdomains applies the ATS exceptions '
+                            'for the given domain to all '
+                            'subdomains as well. '
+                            'For example, the ATS exceptions in the '
+                            'domain exception dictionary apply to {}, '
+                            'as well as math.{}, history.{}, and so on. '
```
[the remark this answers is not in the document]
Comment (rajuraju14) [reply StaticAnalyzer/views/ios/app_transport_security.py]: Thank you!

File: StaticAnalyzer/views/ios/app_transport_security.py [hunk 6]
```diff
@@ -32,46 +32,131 @@ def check_transport_security(p_list):
...
+                            'Otherwise, if the value is NO, the exceptions '
+                            'apply only to '
+                            '{}.'.format(domain, domain, domain, domain)
+                        ),
+                    }
+                    ats.append(findings)
+
+                inc_min_tls = config.get('NSExceptionMinimumTLSVersion', None)
+                if inc_min_tls is not None:
```
[the remark this answers is not in the document]
Comment (rajuraju14) [reply StaticAnalyzer/views/ios/app_transport_security.py]: TLS checks have been added in the latest commit.

File: StaticAnalyzer/views/ios/app_transport_security.py [hunk 7]
```diff
@@ -32,46 +32,131 @@ def check_transport_security(p_list):
...
-                        findings['description'] += (
-                            'Minimum TLS '
-                            'Version: {}'.format(
-                                config.get(
-                                    'NSTemporaryExceptionMinimumTLSVersion'))
-                        )
                     ats.append(findings)
+
+                sec = config.get('NSExceptionRequiresForwardSecrecy', False)
+                if sec is True:
```
[the remark this answers is not in the document]
Comment (rajuraju14) [reply StaticAnalyzer/views/ios/app_transport_security.py]: This has been resolved in the latest commit.

File: StaticAnalyzer/views/ios/app_transport_security.py [hunk 8]
```diff
@@ -32,46 +32,131 @@ def check_transport_security(p_list):
...
+                    findings = {
+                        'issue': ('NSExceptionRequiresForwardSecrecy '
+                                  'set to YES'
+                                  ' for {}'.format(domain)),
+                        'status': 'info',
```
[the remark this answers is not in the document]
Comment (rajuraju14) [reply StaticAnalyzer/views/ios/app_transport_security.py]: This has been resolved in the latest commit.

File: templates/pdf/ios_report.html (hunk 1 above)
Comment (rajuraju14) [reply templates/pdf/ios_report.html]: This has been changed in the latest commit.

File: templates/static_analysis/ios_binary_analysis.html (hunk 2 above)
Comment (rajuraju14) [reply templates/static_analysis/ios_binary_analysis.html]: This has been changed in the latest commit.

File: templates/static_analysis/ios_source_analysis.html (hunk 3 above)
Comment (rajuraju14) [reply templates/static_analysis/ios_source_analysis.html]: This has been changed in the latest commit.

File: StaticAnalyzer/views/ios/app_transport_security.py [hunk 9]
```diff
@@ -32,46 +32,131 @@ def check_transport_security(p_list):
...
+                            'PFS for the given domain. This key is optional. '
+                            'The default value is YES, which limits the '
+                            'accepted ciphers to those that support '
+                            'PFS through Elliptic Curve Diffie-Hellman '
+                            'Ephemeral (ECDHE) key exchange.'),
+                    }
+                    ats.append(findings)
+
+                cert = config.get('NSRequiresCertificateTransparency', False)
+                if cert is True:
```
[the remark this answers is not in the document]
Comment (rajuraju14) [reply StaticAnalyzer/views/ios/app_transport_security.py]: This has been resolved in the latest commit.

File: StaticAnalyzer/views/ios/app_transport_security.py [hunk 10]
```diff
@@ -32,46 +32,160 @@ def check_transport_security(p_list):
...
+                            'as well as math.{}, history.{}, and so on. '
+                            'Otherwise, if the value is NO, the exceptions '
+                            'apply only to '
+                            '{}.'.format(domain, domain, domain, domain)
+                        ),
+                    }
+                    ats.append(findings)
+
+                inc_min_tls = config.get('NSExceptionMinimumTLSVersion', None)
+                if inc_min_tls == 'TLSv1.0' or inc_min_tls == 'TLSv1.1':
```
Comment (ajinabraham) [inline StaticAnalyzer/views/ios/app_transport_security.py]: You can also write something like `if inc_min_tls in ['TLSv1.0', 'TLSv1.1']:`

**Pull request merged**